docs/concepts/select.md

```diff
@@ -354,7 +364,102 @@ export class CityService {
 
 now we can use this `zooThemeParks` selector anywhere in our application.
 
-### Inheriting selectors  
+### The Order of Interacting Selectors
+This topic may be helpful for those users who keep their selectors in separate classes. Let's look at the code below:
+```ts
+// counter.state.ts
+export interface CounterStateModel {
+  counter: number;
+}
+
+@State<CounterStateModel>({
+  name: 'counter',
+  defaults: {
+    counter: 0
+  }
+})
+export class CounterState {}
+
+// counter.query.ts
+export class CounterQuery {
+
+  @Selector([CounterQuery.getCounter])
+  static getCounterCube(counter: number): number {
+    return counter ** 3;
+  }
+
+  @Selector([CounterState])
```

Comment:
It may be worth adding a comment above this line to state the following:
```TS
// Note: this selector being declared after its' usage will cause an issue!!!
```